src/Modera/SecurityBundle/PasswordStrength/StrongPasswordValidator.php

```diff
@@ -57,28 +56,68 @@ public function validate($value, Constraint $constraint)
                 $errorMsg = T::trans($errorMsg);
             }
 
-            $regexConstr = new Regex(array('pattern' => '/[0-9]/'));
-            $regexConstr->message = $errorMsg;
+            $regexConstraint = new Regex(array('pattern' => '/[0-9]/'));
+            $regexConstraint->message = $errorMsg;
 
-            $this->context->setConstraint($regexConstr);
-            $lengthValidator = new RegexValidator();
-            $lengthValidator->initialize($this->context);
-            $lengthValidator->validate($value, $regexConstr);
+            $this->subValidate(
+                new RegexValidator(),
+                $value,
+                $regexConstraint
+            );
         }
 
-        if ($this->config->isCapitalLetterRequired()) {
-            $errorMsg = 'Password must contain at least one capital letter.';
-            if (class_exists('Modera\FoundationBundle\Translation\T')) {
-                $errorMsg = T::trans($errorMsg);
+        if ($this->config->isLetterRequired()) {
+            switch ($this->config->getLetterRequiredType()) {
+                case PasswordConfigInterface::LETTER_REQUIRED_TYPE_CAPITAL_OR_NON_CAPITAL:
+                    $pattern = '/[A-Za-z]/';
+                    $errorMsg = 'Password must contain at least one letter.';
+                    if (class_exists('Modera\FoundationBundle\Translation\T')) {
```

Comment:
As I already said you have 4 very similar blocks here which you can refactor to a parametrized method, something like that:

```
private function trans($message)
{
    return class_exists('Modera\FoundationBundle\Translation\T') ? T::trans($message) : $message;
}
```

Reply:
we cannot do this, because then we cannot import them